source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -503,6 +503,25 @@ const std::string& SslSocketInfo::tlsVersion() const {
   return cached_tls_version_;
 }
 
+const absl::optional<std::string>&
+SslSocketInfo::x509Extension(absl::string_view extension_name) const {
+  auto itr = cached_x509_extensions_.find(extension_name);
+  if (itr != cached_x509_extensions_.end()) {
```

Comment:
Design alternative: should the caller of this function handle caching if it is needed? Or is there a use case where several different filters all need to get the same extension?

Reply:
I don't have a strong opinion, it's easy enough to do caching of this on the caller side. Given that it simplifies things I'll go with that